src/App.tsx

```diff
@@ -13,53 +11,48 @@ import { InlinePlayground } from "./components/InlinePlayground";
 import AppConfig from "./services/configservice";
 import { PLAYGROUND_UI_COLORS } from "./theme";
 import { ThemeProvider } from "@/components/ThemeProvider";
+import {
+  Outlet,
+  RouterProvider,
+  createRouter,
+  createRoute,
+  createRootRoute,
+} from "@tanstack/react-router";
+import { TanStackRouterDevtools } from "@tanstack/react-router-devtools";
 
-export interface AppProps {
-  /**
-   * forTesting indicates whether the app is for testing.
-   */
-  forTesting?: boolean | undefined;
-}
-
-function ForTesting() {
-  return <FullPlayground />;
-}
+const rootRoute = createRootRoute({
+  component: () => (
+    <>
+      <Outlet />
+      <TanStackRouterDevtools />
+    </>
+  ),
+});
 
-function ThemedApp(props: {
-  withRouter?: () => ReactNode;
-  forTesting: boolean | undefined;
-}) {
-  if (window.location.pathname.indexOf("/i/") >= 0) {
-    return (
-      // @ts-expect-error RRv5 types are jank
-      <BrowserRouter>
-        <InlinePlayground />
-      </BrowserRouter>
-    );
-  }
-
-  if (window.location.pathname.indexOf("/e/") >= 0) {
-    return (
-      // @ts-expect-error RRv5 types are jank
-      <BrowserRouter>
-        <EmbeddedPlayground />
-      </BrowserRouter>
-    );
-  }
-
-  if (props.forTesting) {
-    return <ForTesting />;
-  } else {
-    return (
-      // @ts-expect-error RRv5 types are jank
-      <BrowserRouter>
-        <FullPlayground />
-      </BrowserRouter>
-    );
-  }
-}
+// TODO: extend the routing; the $s are catchalls.
+const indexRoute = createRoute({
+  getParentRoute: () => rootRoute,
+  path: "$",
+  component: FullPlayground,
+});
+const inlineRoute = createRoute({
+  getParentRoute: () => rootRoute,
+  path: "/i/$",
+  component: InlinePlayground,
+});
+const embeddedRoute = createRoute({
+  getParentRoute: () => rootRoute,
+  path: "/e/$",
+  component: EmbeddedPlayground,
+});
```

Comment:
This is how you define what they call "code-based routes" (as opposed to "file-based routes"). The docs strongly recommend file-based routing, but that would require more changes to the file structure of the application, so I chose not to do it in this PR. It may be something we want to do in the future.